src/scalar.rs

```diff
@@ -62,6 +85,15 @@ impl IndexMut<usize> for Scalar {
     }
 }
 
+impl Neg for Scalar {
```

Comment:
That's probably faster. Poking at the limbs sounds kind of messy/ugly, but I can't really think of any other reason.